File: packages/gbnfjs/src/rules-builder/parse-name.test.ts
import { describe, it, expect, } from 'vitest';
import { GrammarParseError } from '../utils/errors.js';
import { PARSE_NAME_ERROR, parseName } from './parse-name.js';

describe('parseName', () => {
  it('should return correct position when encountering a valid name', () => {
    const src = 'validName';
    const name = parseName(src, 0);
    expect(name).toEqual(src);
  });

  it('should return correct position when encountering a valid name starting at a non-zero position', () => {
    const src = '123validName';
    const name = parseName(src, 3);
    expect(name).toEqual('validName');
  });

  it('should throw an error when encountering an invalid name', () => {
    const src = '123';
    expect(() => {
      parseName(src, 0);
    }).toThrow(new GrammarParseError(src, 0, PARSE_NAME_ERROR));
  });
});

File: packages/gbnfpy/gbnf/rules_builder/parse_name_test.py
import pytest

from .errors import GrammarParseError
from .parse_name import PARSE_NAME_ERROR, parse_name


def test_should_return_correct_name_when_encountering_a_valid_name():
    src = "validName"
    name = parse_name(src, 0)
    assert name == src


def test_should_return_correct_name_when_encountering_a_valid_name_starting_at_a_non_zero_position():
    src = "123validName"
    name = parse_name(src, 3)
    assert name == "validName"


def test_should_throw_error_when_encountering_an_invalid_name():
    src = "123"
    with pytest.raises(GrammarParseError) as exc_info:
        parse_name(src, 0)
    assert str(exc_info.value) == str(GrammarParseError(src, 0, PARSE_NAME_ERROR))
